File: 2000-2999/2723/OpenJudge2808.py
# coding: utf-8
# -*- coding: utf-8 -*-

L, M = input().split(" ")
L, M = int(L), int(M)
trees = [True]*(L+1)
for m in range(M):
    start, end = input().split(" ")
    start, end = int(start), int(end)
    for i in range(start, end+1):
        trees[i] = False
cnt = 0
for i in trees:
    if i:
        cnt += 1
print(cnt)

File: 2000-2999/2723/OpenJudge2808.java
import java.io.* ;
import java.util.* ;

public class Main {
    public static void main(String[] argv) {
        Scanner cin = new Scanner(System.in) ;
        int L = cin.nextInt() ;
        int M = cin.nextInt() ;

        boolean[] trees = new boolean[L+1] ;
        Arrays.fill(trees, true) ;

        for (int m = 0; m < M; ++m) {
            int start = cin.nextInt() ;
            int end = cin.nextInt() ;

            for (int i = start; i <= end; ++i) {
                trees[i] = false ;
            }
        }

        int cnt = 0 ;
        for (int i = 0; i <= L; ++i) {
            if (trees[i]) {
                ++cnt ;
            }
        }

        System.out.println(cnt) ;
    }
}